examples/medplum-task-example-app/app/src/pages/TaskPage.tsx

```diff
@@ -0,0 +1,105 @@
+import { Grid, Paper, Tabs, Title } from '@mantine/core';
+import { getDisplayString, resolveId } from '@medplum/core';
+import { Patient, Task } from '@medplum/fhirtypes';
+import { DefaultResourceTimeline, Document, ResourceTable, useMedplum } from '@medplum/react';
+import { useEffect, useState } from 'react';
+import { useParams, useNavigate } from 'react-router-dom';
+import { PatientChart } from '../components/PatientChart';
+
+import { TaskActions } from '../components/task-actions/TaskActions';
+import { NotesPage } from './NotesPage';
+
+export function TaskPage(): JSX.Element {
+  const medplum = useMedplum();
+  const navigate = useNavigate();
+  const { id } = useParams() as { id: string };
+  const [task, setTask] = useState<Task | undefined>(undefined);
+  const tabs = ['Details', 'Timeline', 'Notes'];
+  const [patient, setPatient] = useState<Patient | undefined>();
+
+  // Set the current tab to what is in the URL, otherwise default to 'Details'
+  const [currentTab, setCurrentTab] = useState<string>(() => {
+    const tab = window.location.pathname.split('/').pop();
+    return tab && tabs.map((t) => t.toLowerCase()).includes(tab) ? tab : tabs[0].toLowerCase();
+  });
+
+  useEffect(() => {
+    // Fetch the task that is specified in the URL
+    const fetchTask = async (): Promise<void> => {
+      try {
+        const taskData = await medplum.readResource('Task', id);
+        setTask(taskData);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+
+    // Fetch the patient that this task if for. See https://www.medplum.com/docs/careplans/tasks#task-assignment for more details
+    const fetchLinkedPatient = async (): Promise<void> => {
+      if (task?.for) {
+        const patientId = resolveId(task.for);
+        try {
+          const patientData = patientId ? await medplum.readResource('Patient', patientId) : undefined;
+          setPatient(patientData);
+        } catch (error) {
+          console.error(error);
+        }
+      }
+    };
+
+    fetchTask().catch((error) => console.error(error));
+    fetchLinkedPatient().catch((error) => console.error(error));
+  }, [medplum, id, task]);
+
+  // Update the current tab and navigate to its URL
+  const handleTabChange = (newTab: string): void => {
+    setCurrentTab(newTab);
+    navigate(`/Task/${id}/${newTab}`);
+  };
+
+  const onTaskChange = (updatedTask: Task): void => {
+    setTask(updatedTask);
+  };
+
+  if (!task) {
+    return <Document>No Task found</Document>;
+  }
+
+  return (
+    <Grid gutter="xs">
```

Comment:
Minor refactor: Split each panel (Patient profile, center, and actions) into their own components, internal to this file.

WIll make it easier to readh, and will hide all the `Papers` and `Tabs`